Tribler/Core/Modules/MetadataStore/OrmBindings/channel_node.py

```diff
@@ -214,4 +231,50 @@ def from_payload(cls, payload):
         def from_dict(cls, dct):
             return cls(**dct)
 
+        @property
+        @db_session
+        def is_personal(self):
+            return database_blob(self._my_key.pub().key_to_bin()[10:]) == database_blob(self.public_key)
+
+        @db_session
+        def soft_delete(self):
+            if self.status == NEW:
+                # Uncommited metadata. Delete immediately
+                self.delete()
+            else:
+                self.status = TODELETE
+
+        def update_properties(self, update_dict):
+            signed_attribute_changed = False
+            for k, value in update_dict.items():
+                if getattr(self, k) != value:
+                    setattr(self, k, value)
+                    signed_attribute_changed = signed_attribute_changed or (k in self.payload_arguments)
+
+            if signed_attribute_changed:
+                if self.status != NEW:
+                    self.status = UPDATED
+                self.timestamp = self._clock.tick()
+                self.sign()
+
+            return self
+
+        def get_parents_ids(self, recursion_depth=15):
```

Comment:
The name can be a bit misleading, I agree. However, in that case, it would require creating another argument to represent the maximum recursion depth. I'll rename this one to `max_recursion_depth`